### PR TITLE
Implementing yarn workspaces 🚀

### PR DESCRIPTION
**Pull request by evertonfraga**
This PR improves our monorepo package management system by a large margin. There's no need for `bootstrap` commands anymore.

Tasks:
- [x] Migrate this change to Yarn https://github.com/ethereumjs/ethereumjs-vm/pull/884/files
- [x] Fix test runners that use ethereum/tests

### CODE REVIEW

**Comment by evertonfraga** [comment]
Yarn has a problem loading `git submodules` from git dependencies. https://github.com/yarnpkg/yarn/issues/1488

It is causing the failures shown here: https://github.com/ethereumjs/ethereumjs-vm/pull/900/checks?check_run_id=1327571708#step:7:191

**Comment by holgerd77** [comment]
Could you summarize here a bit what problems this is going to solve respectively what properties might be expected from a switch? I've lost a bit track of this.

**Comment by evertonfraga** [comment]
@holgerd77 sure!

The first reason is performance. There are issues with the combo `npm + Lerna`, which you seem to experience daily. Yarn workspaces is a native solution to manage dependencies in a multi-project setup, and it's just incredibly fast. A dirty `lerna bootstrap` takes ~25s, while `yarn` under the same conditions takes only ~2s.

The other reason is to get rid of Lerna. It is highly opinionated and unexpectedly breaks things. I had poor experiences myself. It seems that @alcuadrado and @ricmoo already did the switch and that was additional inspiration :)

**Comment by evertonfraga** [comment]
@holgerd77 I forgot to mention that we'll also get rid of this weird behavior:

> if you run npm install from the package directory, it will remove all links to the local packages, pulling all dependencies from npm. Run npm install from the root only.

**Comment by holgerd77** [comment]
@evertonfraga Ok, I wouldn't dare to just merge this after the merge of #953, please give this an explicit note if you think it is ready for review! 😄

**Comment by holgerd77** [comment]
(there is also still a WIP label on this PR)

**Comment by evertonfraga** [comment]
@holgerd77 right! 

I'd say to only merge this after you're all comfortable with the new submodule functionality, introduced at #953 .

**Comment by holgerd77** [comment]
@evertonfraga I think everyone had the chance to give reaction on #953 and we can now review and merge here once updated (still open for comments though of course until merged).

**Comment by holgerd77** [comment]
Short note: let's please wait on merging this in here until after the final releases on Tuesday. I assume there will be no side effects on this but with such a broad base tooling change you'll never know and I'd rather like to play safe here.

**Comment by holgerd77** [comment]
This can theoretically be integrated now. 😄

**Comment by holgerd77** [comment]
(@evertonfraga please let us know if you want to do yourself or if you want someone else to take this over)

**Comment by evertonfraga** [comment]
@holgerd77 @ryanio hey :)

y'all want me to rebase this branch and put it up for review?

**Comment by holgerd77** [comment]
@evertonfraga Hey ho, happy New Year 🎊, hope you had great holidays!

To your question: Yes. 😄

**Comment by Assawal** [comment]
> This PR improves our monorepo package management system by a large margin. There's no need for `bootstrap` commands anymore.
> 
> Tasks:
> 
> * [x]  Migrate this change to Yarn https://github.com/ethereumjs/ethereumjs-vm/pull/884/files
> * [x]  Fix test runners that use ethereum/tests

**Comment by ryanio** [comment]
Continuing this effort in #1395 with new npm v7 workspaces functionality 👍
